src/_TextInputWrapper.tsx

```diff
@@ -25,39 +43,47 @@ type StyledWrapperProps = {
   hasIcon?: boolean
   block?: boolean
   contrast?: boolean
+  validationStatus?: 'error' | 'warning'
+  /** @deprecated Use `size` prop instead */
   variant?: 'small' | 'large'
+  size?: 'small' | 'large'
 } & WidthProps &
   MinWidthProps &
   MaxWidthProps &
   SxProp
 
 const TextInputWrapper = styled.span<StyledWrapperProps>`
-  display: inline-flex;
-  align-items: stretch;
+  width: max-content;
   min-height: 34px;
   font-size: ${get('fontSizes.1')};
   line-height: 20px;
   color: ${get('colors.fg.default')};
   vertical-align: middle;
+  background-color: ${get('colors.canvas.default')};
   background-repeat: no-repeat; // Repeat and position set for form states (success, error, etc)
   background-position: right 8px center; // For form validation. This keeps images 8px from right and centered vertically.
   border: 1px solid ${get('colors.border.default')};
   border-radius: ${get('radii.2')};
   outline: none;
   box-shadow: ${get('shadows.primer.shadow.inset')};
   cursor: text;
+  padding: 6px 12px;
```

Comment:
i'm afraid there's too much horizontal padding for a text input in here. We should aim for more compact UIs where possible 😇.

> ![Screen Shot 2021-12-06 at 15 22 06](https://user-images.githubusercontent.com/293280/144938594-e44ce574-54fd-4cba-b417-3f07c7172e4d.png)


```suggestion
  padding: 6px 8px;
```

Reply:
Also, because this padding is being set in the wrapper, and not in the `<input>` itself, the surroundings of the input element aren't interactive.

https://user-images.githubusercontent.com/293280/144939127-644cab4b-d115-4cf5-92f1-286ef13d5bcd.mov

We should make sure the `<input>` area covers as much room as possible (saving for leading and trailing elements) within the wrapper.

Reply:
One last thing :)

The vertical padding of `6px` is ignoring the border, resulting in an input field height of 34px[^1]. We should make sure the component height as a whole matches 32px.

🙇 

[^1]: 20px of line height, 6px+6px of vertical padding, 1px+1px of border

Reply:
> The height on that component is a minimum of 34px and the padding is the same as this component.

We should probably fix it in there too. We should make sure the form components match sizes with buttons at a 32px height for the default size variant. c/c @mperrotti

Reply:
@vdepizzol - I'm going to take care of this in a follow-up PR. It's not a trivial amount of work to add to these changes.